metadata-ingestion/src/datahub/ingestion/api/source_helpers.py

```diff
@@ -274,6 +279,56 @@ def auto_browse_path_v2(
         telemetry.telemetry_instance.ping("incorrect_browse_path_v2", properties)
 
 
+def auto_empty_dataset_usage_statistics(
+    stream: Iterable[MetadataWorkUnit],
+    *,
+    dataset_urns: Set[str],
+    config: BaseTimeWindowConfig,
+    all_buckets: bool = False,  # TODO: Enable when CREATE changeType is supported for timeseries aspects
```

Comment:
Whenever we enable this, we should check start_time, end_time defaults in config more carefully. Especially concerned that we may miss out on some usage for the bucket if the UPSERT is disabled.

With current defaults(without stateful ingestion). For today, entire usage until the ingestion time is emitted in today's run and it gets overwritten when usage runs again tomorrow with entire yesterday window.

Reply:
Yeah, this seems like a pretty serious issue. I guess best solution here is to set start and end time correctly, to the start and end time of the nearest bucket